### PR TITLE
Off viewport tabs + favicon

### PR DESCRIPTION
**Pull request by maciejhirsz**
When selecting a network that's off the viewport, it will now show up on the very right with small gradient covering the regular tabs. I've changed the heuristics for the number of tabs displayed to get this working, so you might see fewer of them in normal use (the alternative is to measure when tab is out of viewport in DOM and cause layout trashing in the browser).

Also favicon, went with the plain SVG, can add the white circle underneath if we need it?

image

### CODE REVIEW

File: frontend/src/components/Chains.tsx [hunk 1]
```diff
@@ -27,42 +28,59 @@ import './Chains.css';
...
+      this.props.subscribedHash !== nextProps.subscribedHash ||
       performance.now() - this.lastRender > RENDER_THROTTLE
     );
   }
 
   public render() {
     this.lastRender = performance.now();
+    this.subscribedChainInView = false;
 
-    const allChainsHref = this.props.subscribed
```
Comment (jsdw) [inline frontend/src/components/Chains.tsx:58]: Is there anything that leads to this render method being called again if the viewport width changes?

File: frontend/src/components/Chains.tsx [hunk 2]
```diff
@@ -27,42 +28,59 @@ import './Chains.css';
...
-      ? `#all-chains/${this.props.subscribed}`
+    const viewportWidth = viewport().width;
+    const { chains, subscribedHash, subscribedData } = this.props;
+
+    const renderedChains = chains
+      .slice(0, (viewportWidth / AVERAGE_TAB_WIDTH) | 0)
```
Comment (jsdw) [inline frontend/src/components/Chains.tsx:66]: Did you mean `||` here?

Comment (maciejhirsz) [reply frontend/src/components/Chains.tsx:66]: No, it just forced the number to be an integer at the end.

File: frontend/src/components/Chains.tsx (hunk 1 above)
Comment (maciejhirsz) [reply frontend/src/components/Chains.tsx:58]: It refreshes regularly on each render. I was thinking about adding a `resize` event listener to the component, but it seemed like an overkill.

File: frontend/src/components/Chains.tsx (hunk 2 above)
Comment (jsdw) [reply frontend/src/components/Chains.tsx:66]: Ah I see!

**Review by jsdw — APPROVED** [review]
Love the new favicon, and showing the off screen tab is nice! 

There can be a fairly big gap between tabs on display and the "..." (esp when the selected chain isn't off the list) but I'm ok with us addressing that in a future PR if we are bothered by it (we'd need to do proper measuring I think to really nail it)

**Review by TarikGul — APPROVED** [review]
lgtm, great job!

**Pull request merged**